### PR TITLE
Introducing mesh algorithms

### LINKED ISSUE #4341: [feature] [mesh] fix #31550 Expose new methods in analysis library for exporting contours for mesh layers
Original commit: https://github.com/qgis/QGIS/commit/dd98331ace577da762c05485aa59a4861d7137fe by wonder-sk

The mesh styling dialog now has new combo box for selection of resampling method for datesets defined on faces.
Also the analysis library has new class QgsMeshContours with export function for contour lines and contour polygons.

### PR DESCRIPTION
**Pull request by DelazJ**
This is an initial walk through the mesh algorithms. I'm not a user of this type of data and probably lack some knowledge so there are some blank cells in the description
Let's say that it fixes #4341 (and more unreported)

### CODE REVIEW

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 1]
```diff
@@ -0,0 +1,907 @@
...
+       * 1 --- Defined date/time
+       * 2 --- Dataset group time step
+   * - **Increment between contour levels**
+
+       Optional
+     - ``INCREMENT``
+     - [number]
+
+       Default: *Not set*
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:53]:
```suggestion
     - Interval between generated levels.
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 2]
```diff
@@ -0,0 +1,907 @@
...
+       Default: *Not set*
+     - 
+   * - **Minimum contour level**
+
+       Optional
+     - ``MINIMUM``
+     - [number]
+
+       Default: *Not set*
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:61]:
```suggestion
     - Starting level values of contours.
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 3]
```diff
@@ -0,0 +1,907 @@
...
+       Default: *Not set*
+     - 
+   * - **Maximum contour level**
+
+       Optional
+     - ``MAXIMUM``
+     - [number]
+
+       Default: *Not set*
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:69]:
```suggestion
     - Maximum values of contours, that is no generated levels will be greater that this value.
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 4]
```diff
@@ -0,0 +1,907 @@
...
+       Default: *Not set*
+     - 
+   * - **List of contours level**
+
+       Optional
+     - ``MINIMUM``
+     - [number]
+
+       Default: *Not set*
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:77]:
```suggestion
     - List of wanted levels of contours (separated by commas). If filled, the increment, minimum, and maximum field will not be considered.
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 5]
```diff
@@ -0,0 +1,907 @@
...
+  :start-after: **algorithm_code_section**
+  :end-before: **end_algorithm_code_section**
+
+
+.. _qgistinmeshcreation:
+
+TIN mesh creation
+-----------------
+
+Creates a TIN mesh layer from vector layers.
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:821]:
```suggestion
Creates a TIN mesh layer from vector layers. The TIN mesh is created using a Delaunay triangulation.
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 6]
```diff
@@ -0,0 +1,907 @@
...
+     - GUI ONLY
+     - [tablefield: any]
+     - A selector of the field to use from the selected layer.
+       Each vertex is assigned the corresponding value of its original feature.
+   * - **Use Z-coordinate for value on vertex**
+     - GUI ONLY
+     - [boolean]
+
+       Default: False
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:853]:
```suggestion
     - If checked, the Z value of vector layer points or polygons/lines vertices will be used to assign the Z value of the vertex mesh layer. Only available if the input layers are 3D.
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 7]
```diff
@@ -0,0 +1,907 @@
...
+     - The time range to take into account
+
+       * 0 --- Current canvas time
+       * 1 --- Defined date/time
+       * 2 --- Dataset group time step
+   * - **Lines for data export**
+     - ``INPUT_LINES``
+     - [vector: line]
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:180]:
```suggestion
     - Lines where the data will be extracted from the dataset mesh
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 8]
```diff
@@ -0,0 +1,907 @@
...
+   * - **Line segmentation resolution**
+     - ``RESOLUTION``
+     - [number]
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:184]:
```suggestion
     - The distance between points on the lines where the data will be extracted from the dataset mesh.
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 9]
```diff
@@ -0,0 +1,907 @@
...
+   * - **Output coordinate system**
+
+       Optional
+     - ``CRS_OUTPUT``
+     - [crs]
+     - Coordinate Reference System to assign to the output
+   * - **Export vector option**
+     - ``VECTOR_OPTION``
+     - [enumeration]
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:275]:
```suggestion
     - Coordinate type of vector value exportation: cartesian, polar or both.
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 10]
```diff
@@ -0,0 +1,907 @@
...
+     - [number]
+
+       Default: 2
+     -
+   * - **Points for data export**
+     - ``INPUT_POINTS``
+     - [vector: point]
+
+       Default: 2
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:665]:
```suggestion
     - Vector layer containing points where the data will be extracted from the dataset mesh
```

File: docs/user_manual/processing_algs/qgis/mesh.rst [hunk 11]
```diff
@@ -0,0 +1,907 @@
...
+       .. include:: ../algs_include.rst
+          :start-after: **extent_options**
+          :end-before: **end_extent_options**
+
+   * - **Pixel size**
+     - ``PIXEL_SIZE``
+     - [number]
+
+       Default: 1.0
+     -
```
Comment (vcloarec) [inline docs/user_manual/processing_algs/qgis/mesh.rst:771]:
```suggestion
     - Pixel size of the output raster layer.
```

**Review by vcloarec — CHANGES_REQUESTED** [review]

**Review by vcloarec — APPROVED** [review]

**Comment by DelazJ** [comment]
Thanks @vcloarec

**Pull request merged**